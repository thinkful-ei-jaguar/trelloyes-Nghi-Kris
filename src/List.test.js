import React from 'react';
import reactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import List from './List';

// Props
// header = string
// cards = array of objects with id, title, and content

// Snapshot Test
it('List UI is consistent', () => {
    const deck = [{id:'1', title:'Ace', content:'Red Spade Ace'}, {id:'2', title:'Queen', content:'Black Heart Queen'}];
    const jsonFile = renderer.create(<List header="Deck of Cards" cards={deck} />).toJSON();
    expect(jsonFile).toMatchSnapshot();
})

// cards => array with object and each object has an id, title, and
// Smoke Test
it('Rendering List',() => {
    const div = document.createElement('div');
    reactDOM.render(<List />, div);
    reactDOM.unmountComponentAtNode();
})