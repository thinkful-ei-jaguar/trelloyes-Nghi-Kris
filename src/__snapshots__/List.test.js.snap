// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List UI is consistent 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    Deck of Cards
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        Ace
      </h3>
      <p>
        Red Spade Ace
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Queen
      </h3>
      <p>
        Black Heart Queen
      </p>
    </div>
     
  </div>
</section>
`;
